File: Frontend/src/app/core/api/models/post-dto.ts
/* tslint:disable */
/* eslint-disable */
import { CommentDto } from './comment-dto';
import { IngredientDto } from './ingredient-dto';
import { PreparationStepDto } from './preparation-step-dto';
import { ProductCategoryDto } from './product-category-dto';
import { UserDto } from './user-dto';
export interface PostDto {
  author?: UserDto;
  categories?: Array<ProductCategoryDto>;
  comments?: Array<CommentDto>;
  cookingTime?: number;
  created?: string;
  difficulty?: number;
  ingredients?: Array<IngredientDto>;
  isLikedByUser?: boolean;
  numberOfLikes?: number;
  postId?: number;
  postPicture?: string;
  preparationSteps?: Array<PreparationStepDto>;
  shortDescription?: string;
  title?: string;
  updated?: string;
}

File: Projekt/Backend/tasty/src/main/java/pl/poli/paint/tasty/dto/PostDTO.java
package pl.poli.paint.tasty.dto;

import lombok.Builder;
import lombok.Data;

import java.util.Date;
import java.util.List;

@Data
@Builder
public class PostDTO {

    private Long postId;
    private String title;
    private List<PreparationStepDTO> preparationSteps;
    private List<ProductCategoryDTO> categories;
    private List<IngredientDTO> ingredients;
    private List<CommentDTO> comments;
    private String shortDescription;
    private Integer difficulty;
    private Short cookingTime;
    private UserDTO author;
    private Date created;
    private Date updated;
    private String postPicture;
    private Boolean isLikedByUser;
    private Integer numberOfLikes;
}
